### ISSUE TITLE
Phonological metrics mask padding with the orthographic pad id: every phon metric scores padded positions

### ISSUE THREAD

**Issue by mathnathan** [emelex-ai/BRIDGE#227]
## Summary

Every metric in `bridge/application/training/phon_metrics.py` masks out padding with
`phon_true != 2`. Phonological targets never contain the value `2`. `2` is the *orthographic*
pad id; the phonological one is `35`. The mask is therefore always all-`True`, and padded
positions are scored as if they were real phonemes in every phonological metric.

Long-standing, not a regression: the pad fill value is identical on `5d7e8d9`.

## Evidence

```python
from bridge.domain.tokenizer import BridgeTokenizer
tok = BridgeTokenizer()
t = tok.encode(["cat", "hello", "elephant"]).phonological.phon_targets

sorted(int(v) for v in t.unique())            # [0, 1, 35]
tok.phon_pad_id                               # 35   <- phonological [PAD]
tok.orth_pad_id                               # 2    <- orthographic [PAD]

(t != 2).float().mean()                       # 1.0    mask keeps 100% of positions
(t == tok.phon_pad_id).all(-1).float().mean() # 0.292  but 29.2% of positions ARE padding
```

The same `{0, 1, 35}` value set appears in the committed fixture
`tests/application/training/data/phon_true.pt`.

## Where the value comes from

`bridge/domain/tokenizer/phoneme_tokenizer.py:96-97` builds the target table by dropping the
`[PAD]` column from the multi-hot matrix and filling the `[PAD]` *row* with
`special_token_dims["[PAD]"]`, which is `35`:

```python
target_table = phoneme_table.multihot[:, :-1]
target_table[pad_row] = special_token_dims["[PAD]"]
```

That value is deliberate and correct: it matches the `ignore_index` passed to
`CrossEntropyLoss` at `bridge/application/training/training_pipeline.py:134`, so the **loss**
does correctly skip padded positions. Only the metrics use the wrong sentinel.

## Affected metrics

All six masks in the file, at lines 8, 38, 56, 76, 102 and 126:

| line | metric |
|---|---|
| 8 | `calculate_phon_word_accuracy` inner predicate |
| 38 | phoneme feature accuracy |
| 56 | phoneme-wise accuracy |
| 76 | closest-phoneme accuracy |
| 102 | euclidean / cosine distance |
| 126 | word accuracy mask |

Every one of them currently counts padded positions. Because padded rows are entirely `35`
while predictions there are `0`/`1`, those positions score as wrong. Reported accuracies are
therefore biased **downward**, and the size of the bias tracks how much padding a batch has,
so it varies with batch composition. A batch of uniformly long words looks better than a batch
of mixed lengths for reasons that have nothing to do with the model.

## Fix

Replace the literal with the real phonological pad id. It is already reachable:
`Model.model_config.vocab.phon_pad_id`, and `calculate_phon_metrics` already receives enough
context to thread it, exactly as `calculate_orth_metrics` already takes `orth_pad_id`
explicitly (`bridge/application/training/ortho_metrics.py:31`). Mirroring that signature is the
smallest change and removes the magic number:

```python
def calculate_phon_metrics(logits, phonology, phon_reps, phon_pad_id: int) -> dict[str, float]:
    ...
    mask = true != phon_pad_id
```

Note that fixing this **will move every reported phonological metric**, upward, on every run.
That is the correct direction, but it means historical metric rows are not comparable across
the fix, so it is worth landing deliberately rather than quietly.

A regression test should assert that a batch containing at least one short and one long word
produces a mask that excludes the padded positions, e.g. that
`mask.float().mean() < 1.0` for `["a", "elephant"]`.

## Related

- #225 (orthographic loss and metrics target off-by-one) touches the sibling metrics module
- #226 (`load_model` discards the resume epoch)

**Comment by mathnathan** [comment]
Fixed in `130f2b4`, on `main` via #230.

Left open by mistake: #230's body said `Closes #223, #224, #225, #226, #227`, and GitHub only
parses a closing keyword attached to a single reference, so only #223 closed.

`calculate_phon_metrics` takes `phon_pad_id: int` as a required argument, mirroring
`calculate_orth_metrics`, and the six masking helpers take it too. Required rather than
defaulted: a default would let a caller silently reintroduce the defect, and every caller
already holds the value on its config.

The size of the error was measured, not estimated, by loading the real `phon_metrics.py`
source twice and substituting only the sentinel. **The instrument was validated first** on a
batch of equal-length words where no padding exists and the two builds must agree: zero
differences across all eight metrics. Against the committed fixtures, which turn out to be
**54.2% padding**:

| metric | reported | corrected | ratio |
|---|---|---|---|
| `phon_euclidean_distance` | 112.257 | **0.234** | 0.002 |
| `closest_phoneme_l2_accuracy` | 0.3894 | **0.8504** | 2.18 |
| `closest_phoneme_cosine_accuracy` | 0.4483 | **0.9738** | 2.17 |
| `phon_cosine_similarity` | 0.5262 | **0.9487** | 1.80 |

**Both filed descriptions of the direction were half right.** This issue predicted accuracies
biased downward, which holds: padded rows are entirely `35` while predictions there are `0`/`1`,
so padding scores as wrong. #224 predicted everything inflated, which holds for the similarity
and distance metrics, because padded rows are near-identical to each other. `phon_euclidean_distance`
was off by a factor of roughly 480 and was essentially measuring the padding.

This issue's warning about comparability was, if anything, understated. Metric rows recorded
before `130f2b4` are not merely incomparable across the fix; they cannot be corrected after the
fact, because the per-batch padding fraction was never recorded. Treat every phonological number
BRIDGE has reported to date as void.

The pinned values in `tests/application/training/test_phon_metrics.py` moved with it. The
fixtures are untouched and both readings are recorded in the file, so the move is auditable
rather than a silently regenerated baseline. The four identity cases, a tensor compared against
itself, are unmoved to 1e-6, which is what shows the change touched the sentinel and nothing else.

One fragility noted in the code rather than fixed: the masks are elementwise while the reshape
that follows assumes whole rows survive. That holds because a padded target row is entirely
`phon_pad_id` and a real one holds only 0 and 1, verified on the fixtures, and a target encoding
that broke the property would raise from the reshape rather than return a wrong number.

Full numbers in `docs/decisions/0005-phonological-metrics-take-the-pad-id.md`.
